src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -1,19 +1,74 @@
 package core.basesyntax.impl;
 
 import core.basesyntax.Storage;
+import java.util.Objects;
 
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int MAX_ELEMENTS = 10;
+    private static final int MIN_INDEX = 0;
+    private KeyValue<K, V>[] keysValues;
+    private int size;
+
+    public StorageImpl() {
+        keysValues = new KeyValue[MAX_ELEMENTS];
+    }
+
     @Override
     public void put(K key, V value) {
+        int keyIndex = getKeyIndex(key);
+        if (keyIndex >= MIN_INDEX) {
+            keysValues[keyIndex].setValue(value);
+        } else {
+            try {
+                keysValues[size++] = new KeyValue<>(key, value);
+            } catch (RuntimeException e) {
+                throw new RuntimeException("You can't put at the storage more than "
+                        + MAX_ELEMENTS + " elements");
+            }
+        }
     }
 
     @Override
     public V get(K key) {
-        return null;
+        int keyIndex = getKeyIndex(key);
+        return keyIndex >= MIN_INDEX ? keysValues[keyIndex].getValue() : null;
     }
 
     @Override
     public int size() {
+        return size;
+    }
+
+    public int getKeyIndex(K key) {
+        for (int i = 0; i < size; i++) {
+            K currentKey = keysValues[i].getKey();
+            if (Objects.equals(key, currentKey) || currentKey != null
```

Comment:
```suggestion
            if (key == currentKey || currentKey != null
```
[Don’t use class Objects](https://mate-academy.github.io/jv-program-common-mistakes/java-core/generics/storage.html#dont-use-class-objects)